ADR/partner-tasks.md

```diff
@@ -0,0 +1,34 @@
+# Partner Tasks in Build/Test Pipelines
+
+Date: 2023-03-06
+
+## Status
+
+Provisional
+
+## Context
+
+* As a Red Hat Partner, I would like to offer our service's capability as a Tekton Task that would be executed in a user's build/test Pipeline on StoneSoup.
+* As a StoneSoup maintainer, I would like to provide a way to ensure the above Tekton Task is validated and approved for use in the build/test Pipeline on StoneSoup.
+
+Associated business ask
+https://issues.redhat.com/browse/STONE-549
+
+## Decision
+
+### Plumbing
+
+1. Setup a new Git repository.
+2. Define a directory structure for Task submissions as manifests in the yaml format.
+3. Configure a CI job that validates the Tasks upon opening of a pull request.
+4. Configure a CI job that generates an OCI artifact consumable in a Tekton Pipeline.
+
+### Day-to-day operations
```

Comment:
How do users discover those pipeline tasks and enable them?

Reply:
Moving a comment in a later thread here as it is more relevant here.

> Would the tasks be injected into the pipeline by default? If so, this will produce many extraneous "skipped" tasks in the UI, especially as the number of partner tasks increases.

Reply:
Technically, this is independent of the "day-to-day" operations described here. But as written above, using the correct organization, it could be easily indexed by artifacthub or something else, and/or pull in our TKNECO catalog, …

Reply:
@gbenhaim , after adding the Task to Red Hat's certification repo ( the short-term/long-term one ),  this steps get the Task into the stonesoup system:

> Manually/Automatically, a PR is opened to the https://github.com/redhat-appstudio/build-definitions repository.

Additionally, the user may have to configure tokens to be able to _not skip_ the Task.

Reply:
"Upon approval from the Build/Test team, the changes are merged.
4. Manually/Automatically, a PR is opened to the https://github.com/redhat-appstudio/build-definitions repository."